[Kandinsky] Improve kandinsky API a bit (#3636)
huggingface/diffusers: docs/source/en/api/pipelines/kandinsky.mdx

# add all the conditions we want to interpolate, can be either text or image
images_texts = ["a cat", img1, img2]

# specify the weights for each condition in images_texts
weights = [0.3, 0.3, 0.4]

# We can leave the prompt empty
prompt = ""
prior_out = pipe_prior.interpolate(images_texts, weights)

pipe = KandinskyPipeline.from_pretrained("kandinsky-community/kandinsky-2-1", torch_dtype=torch.float16)
pipe.to("cuda")

image = pipe(prompt, **prior_out, height=768, width=768).images[0]

image.save("starry_cat.png")
```
![img](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/kandinsky-docs/starry_cat.png)

> ## Optimization

Author: @radames this should help a bit with optimization

Running Kandinsky in inference requires running both a first prior pipeline: [`KandinskyPriorPipeline`]
and a second image decoding pipeline which is one of [`KandinskyPipeline`], [`KandinskyImg2ImgPipeline`], or [`KandinskyInpaintPipeline`].

The bulk of the computation time will always be the second image decoding pipeline, so when looking 
into optimizing the model, one should look into the second image decoding pipeline.

When running with PyTorch < 2.0, we strongly recommend making use of [`xformers`](https://github.com/facebookresearch/xformers)
to speed-up the optimization. This can be done by simply running:

```py
from diffusers import DiffusionPipeline
import torch

t2i_pipe = DiffusionPipeline.from_pretrained("kandinsky-community/kandinsky-2-1", torch_dtype=torch.float16)
t2i_pipe.enable_xformers_memory_efficient_attention()
```

When running on PyTorch >= 2.0, PyTorch's SDPA attention will automatically be used. For more information on 
PyTorch's SDPA, feel free to have a look at [this blog post](https://pytorch.org/blog/accelerated-diffusers-pt-20/).

> To be user, you can also manually set the pipeline to use PyTorch's 2.0 efficient attention:

Reviewer: ```suggestion
To have explicit control , you can also manually set the pipeline to use PyTorch's 2.0 efficient attention:
```

```py
from diffusers.models.attention_processor import AttnAddedKVProcessor2_0

t2i_pipe.unet.set_attn_processor(AttnAddedKVProcessor2_0())
```

> The slowest and most memory intense attention processor is the default `AttnAddedKVProcessor` processor.

Reviewer: did I miss something here? why are we using `AttnAddedKVProcessor` and`AttnAddedKVProcessor2_0`  instead of `AttnProcessor` and `AttnProcessor2_0`?

Author: Because that's the attention processor Kandinsky uses?! We cannot set it to another attention processor type...maybe we need to change the API a bit here going forward

Reviewer: ohh I didn't realize that 😅 I think it's a pretty low-level abstraction that the user should not need to know but then again we only need to explicitly enable it for testing so maybe no need to change the API
